src/salesforce/salesforce-api.ts

```diff
@@ -134,27 +133,31 @@ export class SalesforceApi {
     bodyParams.append('username', this.config.salesforceUsername!);
     bodyParams.append('password', this.config.salesforcePassword!);
 
-    const response = await fetch(`${processedLoginUrl}/services/oauth2/token`, {
+    const url = `${processedLoginUrl}/services/oauth2/token`;
+    const response = await fetch(url, {
       method: 'POST',
       body: bodyParams,
       headers: {
         'Content-Type': 'application/x-www-form-urlencoded',
       },
     });
 
-    if (!response.ok) {
-      const message = JSON.stringify(await response.json());
-      return Promise.reject(
-        new InvalidCredentialsError(
-          `Failed to get Salesforce bearer token. Response: ${message}`,
-          { status: response.status, response: message },
-        ),
+    let data;
+    try {
+      data = await readResponse<SalesforceAccessTokenResponse>(url, response);
+    } catch (apiError) {
+      if (apiError instanceof ApiError) {
+        throw new InvalidCredentialsError(
+          `Failed to get Salesforce bearer token. Reason: ${apiError.message}`,
+          apiError.getDetails(),
+        );
+      }
+      throw new InvalidCredentialsError(
+        `Failed to get Salesforce bearer token. Reason: ${apiError}`,
```

Comment:
If it can be something else than api error maybe just call it error?

Reply:
renamed to error